components/Conversation/Conversation.tsx

```diff
@@ -11,40 +13,40 @@ type ConversationProps = {
 const Conversation = ({
   recipientWalletAddr,
 }: ConversationProps): JSX.Element => {
-  const messagesEndRef = useRef(null)
+  const conversations = useAppStore((state) => state.conversations)
+  const selectedConversation = conversations.get(recipientWalletAddr)
+  const conversationKey = getConversationKey(selectedConversation)
 
-  const scrollToMessagesEndRef = useCallback(() => {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    ;(messagesEndRef.current as any)?.scrollIntoView({ behavior: 'smooth' })
-  }, [])
+  const { sendMessage } = useSendMessage(selectedConversation)
 
-  const { sendMessage } = useConversation(
-    recipientWalletAddr,
-    scrollToMessagesEndRef
+  const [endTime, setEndTime] = useState<Map<string, Date>>(new Map())
+
+  const { convoMessages: messages, hasMore } = useGetMessages(
+    conversationKey,
+    endTime.get(conversationKey)
   )
 
-  const convoMessages = useAppStore((state) => state.convoMessages)
   const loadingConversations = useAppStore(
     (state) => state.loadingConversations
   )
 
-  const messages = useMemo(
-    () => convoMessages.get(recipientWalletAddr) ?? [],
-    [convoMessages, recipientWalletAddr]
-  )
-
-  const hasMessages = messages.length > 0
+  const fetchNextMessages = useCallback(() => {
+    if (
+      hasMore &&
+      Array.isArray(messages) &&
+      messages.length > 0 &&
+      conversationKey
+    ) {
+      const lastMsgDate = messages[messages.length - 1].sent
+      const currentEndTime = endTime.get(conversationKey)
+      if (!currentEndTime || lastMsgDate <= currentEndTime) {
+        endTime.set(conversationKey, lastMsgDate)
+        setEndTime(new Map(endTime))
+      }
+    }
+  }, [conversationKey, hasMore, messages, endTime])
 
-  useEffect(() => {
-    if (!messages || !messagesEndRef.current) return
-    setTimeout(() => {
-      scrollToMessagesEndRef()
-    }, 1000)
-  }, [recipientWalletAddr, scrollToMessagesEndRef, messages])
-
-  if (!recipientWalletAddr) {
-    return <div />
-  }
+  const hasMessages = Number(messages?.length) > 0
```

Comment:
If messages is undefined this is going to resolve to `NaN > 0`. Which also happens to be false, but I think would be better expressed as `(messages?.length ?? 0) > 0`